pkg/registry/service/rest.go

```diff
@@ -144,6 +144,13 @@ func (rs *REST) Delete(ctx api.Context, id string) (runtime.Object, error) {
 		return nil, err
 	}
 
+	// TODO: can leave dangling endpoints, and potentially return incorrect
```

Comment:
I separated this PR into 2 commits.
Basically, the first commit is moving some logic (that is not related to storage operations)
from: pkg/registry/etcd/etcd_test.go
to: pkg/registry/service/rest.go

So I'm not adding it - I'm just moving it to a better location (this is just refactoring, not changing logic).